server/src/main.rs

```diff
@@ -39,25 +48,154 @@ impl CargoRpc for CargoGrpcImpl {
     }
 }
 
-/// Tokio signal handler that will wait for a user to press CTRL+C.
-/// We use this in our hyper `Server` method `with_graceful_shutdown`.
+/// Starts the grpc server for this microservice
+async fn grpc_server() {
+    // GRPC Server
+    let grpc_port = std::env::var("DOCKER_PORT_GRPC")
+        .unwrap_or_else(|_| "50051".to_string())
+        .parse::<u16>()
+        .unwrap_or(50051);
+
+    let addr = format!("[::1]:{grpc_port}").parse().unwrap();
```

Comment:
`[::1]` will break in docker (at least it did for me) you can just use `[::]` and it will bind to all local addresses (ipv4 and ipv6)

Reply:
Fixed